src/core/api.c

```diff
@@ -111,8 +111,20 @@ MsQuicConnectionClose(
 #pragma prefast(suppress: __WARNING_25024, "Pointer cast already validated.")
     Connection = (QUIC_CONNECTION*)Handle;
 
+    BOOLEAN IsWorkerThread = Connection->WorkerThreadID == CxPlatCurThreadID();
+    CXPLAT_TEL_ASSERT(!Connection->State.Freed);
+
+    if (IsWorkerThread && Connection->State.HandleClosed) {
+        //
+        // Close being called from worker thread after being closed by app
+        // thread. This is an app programming bug, and they should be checking
+        // the AppCloseInProgress flag, but as the handle is stil valid here
+        // we can just make this a no-op.
+        //
+        goto Error;
+    }
+
     QUIC_CONN_VERIFY(Connection, !Connection->State.HandleClosed);
```

Comment:
I at least want the CONN_VERIFY first. While we might be able to ignore this in PROD, when they have app verifier enabled, I want to tell them they did bad.

Reply:
Ok. Done.